src/components/SelectNumber/index.tsx

```diff
@@ -0,0 +1,354 @@
+import { Theme, css } from '@emotion/react'
+import styled from '@emotion/styled'
+import PropTypes from 'prop-types'
+import {
+  ChangeEventHandler,
+  FocusEventHandler,
+  InputHTMLAttributes,
+  KeyboardEventHandler,
+  MutableRefObject,
+  ReactNode,
+  useRef,
+  useState,
+} from 'react'
+import parseIntOr from '../../helpers/numbers'
+import Box from '../Box'
+import Icon from '../Icon'
+import Tooltip from '../Tooltip'
+import Touchable from '../Touchable'
+
+const bounded = (value: number, min: number, max: number) =>
+  Math.max(min, Math.min(value, max))
+
+const roundStep = (value: number, step: number, direction: number) =>
+  direction === -1
+    ? Math.floor(value / step) * step
+    : Math.ceil(value / step) * step
+
+const disabledStyles = ({
+  disabled,
+  theme,
+}: {
+  disabled: boolean
+  theme: Theme
+}) =>
+  disabled &&
+  `
+    background-color: ${theme.colors.neutral.backgroundWeakDisabled};
+    border: none;
+    color: ${theme.colors.neutral.textWeakDisabled};
+    opacity: 1;
+    cursor: not-allowed;
+  `
```

Comment:
As this is only used once this can be inlined
This would have shown that we check twice the disabled condition :)